Update k8s api group versions in helm charts

sankar-palanivel opened the pull request:
<!--
Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
Any pull request must pass the automated Travis tests and, if applicable, code style checks. In addition the pull request must  contain tests that cover the code.
-->

### Applicable Issues
#72 

### Description of the Change
The Eiffel easy2use installation on local minikube cluster is failing because of unsupported API group versions defined in the helm charts used in this deployment.

This PR updates,

- The API group versions in the helm charts used for Eiffel easy2use deployment on k8s
- Some of the resource parameter changes as per below entries,

> * `Ingress` and `IngressClass` resources have graduated to `networking.k8s.io/v1`.
* `spec.backend` -> `spec.defaultBackend`
* `serviceName` -> `service.name`
* `servicePort` -> `service.port.name` (for string values)
* `servicePort` -> `service.port.number` (for numeric values)
* `pathType` no longer has a default value in v1, it must be specified
* `path` is no longer required to be a valid regular expression

### Alternate Designs
N/A

### Benefits
This will allow users to deploy the Eiffel easy2use sandbox setup on local minikube k8s clsuter

### Possible Drawbacks
N/A

### Sign-off
<!-- Sign the below certificate of origin, using your full name and e-mail address. -->
<!-- The certificate is copied from https://developercertificate.org/ -->

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: Sankar Palanivel sankar.palanivel@est.tech

m-linner-ericsson reviewed (commented):
Maybe I have misunderstood something but shouldn't we step the version number in the file name if we change these files?

sankar-palanivel commented:
> Maybe I have misunderstood something but shouldn't we step the version number in the file name if we change these files?

I didn't update the version number in these helm chart file namess, as I didn't change the version of Eiffel, Jenkins, MongoDB & RabbitMQ packages used in these charts.
- Updated the k8s API version used for the `kind: Deployment` from `extensions/v1beta1` to `apps/v1` in `jenkins-0.18.0.tgz`
- Updated the k8s API version used for the `kind: Ingress` from `extensions/v1beta1` to `networking.k8s.io/v1` and some of its parameter related to the API version change in `eiffel-2.0.0.tgz`
- Updated the k8s API version used for the `kind: PodDisruptionBudget` from `policy/v1beta1` to `policy/v1` in `mongodb-9.3.1.tgz`
- Updated the k8s API version used for the `kind: PodDisruptionBudget` from `policy/v1beta1` to `policy/v1` in `rabbitmq-7.7.1.tgz`

tobiasake reviewed (commented):
Does Travis-CI support some kind of Kubernetes test deployment that can used as test for each Eiffel Easy2Use PR that is created or something?, e.g. using Minikube that can be added and it would be easier for the reviewer to know that the change in PR works.

m-linner-ericsson commented:
> > Maybe I have misunderstood something but shouldn't we step the version number in the file name if we change these files?
> 
> I didn't update the version number in these helm chart file namess, as I didn't change the version of Eiffel, Jenkins, MongoDB & RabbitMQ packages used in these charts.
> 
> * Updated the k8s API version used for the `kind: Deployment` from `extensions/v1beta1` to `apps/v1` in `jenkins-0.18.0.tgz`
> * Updated the k8s API version used for the `kind: Ingress` from `extensions/v1beta1` to `networking.k8s.io/v1` and some of its parameter related to the API version change in `eiffel-2.0.0.tgz`
> * Updated the k8s API version used for the `kind: PodDisruptionBudget` from `policy/v1beta1` to `policy/v1` in `mongodb-9.3.1.tgz`
> * Updated the k8s API version used for the `kind: PodDisruptionBudget` from `policy/v1beta1` to `policy/v1` in `rabbitmq-7.7.1.tgz`

I would still think we should step the versions as we have changed the helm charts. Maybe just step the minor version?
